src/store/flow/block-types/Core_SetGroupMembershipStore.ts

```diff
@@ -0,0 +1,61 @@
+import { ActionTree, GetterTree, MutationTree } from 'vuex'
+import { IRootState } from '@/store'
+import {
+  IBlockExit,
+  ISetGroupMembershipBlockConfig,
+} from '@floip/flow-runner'
+import { IdGeneratorUuidV4 } from '@floip/flow-runner/dist/domain/IdGeneratorUuidV4'
+import { defaults } from 'lodash'
+import { IFlowsState } from '../index'
+
+export const ADD_KEY = 'add'
+export const REMOVE_KEY = 'remove'
+
+export const BLOCK_TYPE = 'Core\\SetGroupMembership'
+
+export const getters: GetterTree<IFlowsState, IRootState> = {}
+
+export const mutations: MutationTree<IFlowsState> = {}
+
+export const actions: ActionTree<IFlowsState, IRootState> = {
+  async createWith({ dispatch }, { props }: { props: { uuid: string } & Partial<ISetGroupMembershipBlockConfig> }) {
+    const exits: IBlockExit[] = [
+      await dispatch('flow/block_createBlockDefaultExitWith', {
+        props: ({
+          uuid: (new IdGeneratorUuidV4()).generate(),
+          tag: 'Default',
+          label: 'Default',
+        }) as IBlockExit,
+      }, { root: true }),
+    ]
+
+    return defaults(props, {
```

Comment:
It doesn't look obvious here: do you think it matters that `config` property below will always be overwritten in its entirety if provided as part of props argument into this function? 

Like, if defaults has `{config: {groupKey, groupName, isMember}}`, and I provide props as `{config: {isMember: true}}`, then the data returned will have `config` as `{isMember: true}` and without the other properties, because defaults is shallow. 

An option to consider: use `defaultsDeep(…)`

Is it important? 
Does it need to be standardized across `createWith()`s so that behaviour is predictable?

Reply:
This is a very good point! :100: 
I think we need to standardized across all `createWith()`, as both of us understand the impact and are in the same page, I will make changes right in this PR to avoid making new ticket.